src/processor/market.rs

```diff
@@ -0,0 +1,76 @@
+use solana_program::{
+    account_info::AccountInfo, borsh1::try_from_slice_unchecked, entrypoint::ProgramResult,
+    program_error::ProgramError, pubkey::Pubkey, system_program,
+};
+
+use crate::{instruction::CreateMarketArgs, loaders::*, state::Market, utils::*, MARKET};
+use borsh::{BorshDeserialize, BorshSerialize};
+
+// process_init_market creates a new tradable market.
+pub fn process_init_market<'a, 'info>(
+    _program_id: &Pubkey,
```

Comment:
since we have `declare!('')` for our program address, we can use that instead of passing the variable reference here. Safer and we can avoid dangling references.